Fix to adjusting bundle and bitstream policies (item install, embargo lift)

kshepherd opened the pull request:
Apply DEFAULT_ITEM_READ to bundles, not DEFAULT_BITSTREAM_READ so that files can be listed if the item / default item is readable

## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes https://github.com/DSpace/DSpace/issues/8902
* Related to a similar merged PR: https://github.com/DSpace/DSpace/pull/8783

## Description
The `adjustBundleBitstreamPolicies` method in ItemService which is called when installing a new item or lifting a legacy embargo, applies DEFAULT_BITSTREAM_READ to both bundles and bitstreams. As per recent discussion on https://github.com/DSpace/DSpace/issues/8902 we think DEFAULT_ITEM_READ should be used instead.

This PR might not yet be complete:
* I expect to have broken at least one test (this is good, will prove the changed functionality, and I will update / add a new test soon).
* I think we should possibly allow for a missing DEFAULT_ITEM_READ to fall back to inheriting Item READ? (more relevant for an embargo lift than a new install, probably)
* I added a TODO item to ask if we should throw an exception for a missing DEFAULT_ITEM_READ (sort of conflicting with the question above)

## Instructions for Reviewers
Try to reproduce the behaviour described in https://github.com/DSpace/DSpace/issues/8902 with this PR applied


## Checklist

- [ ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [ ] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [ ] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

tdonohue commented on dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java:
```diff
@@ -930,16 +930,23 @@ public void inheritCollectionDefaultPolicies(Context context, Item item, Collect
+        List<ResourcePolicy> defaultCollectionBundlePolicies = authorizeService
+                .getPoliciesActionFilter(context, collection, Constants.DEFAULT_ITEM_READ);
+        // Bitstreams should inherit from DEFAULT_BITSTREAM_READ
+        List<ResourcePolicy> defaultCollectionBitstreamPolicies = authorizeService
```
@kshepherd : Since, it appears all tests still succeed with this change... could you add a *new test* which verifies this behavior?  Basically a test which creates an Item which inherits both `DEFAULT_ITEM_READ` and `DEFAULT_BITSTREAM_READ` from its Collection and verify that the Bundle inherits the former, while the Bitstream inherits the latter.  (It's possible you could modify an existing test to simply add in the Bundle check... I suspect we just don't have any tests to verify how a Bundle inherits permissions.)

tdonohue replied:
Added this test for you in https://github.com/DSpace/DSpace/pull/8901/commits/804fce1c99419b29e745119460f3c351a8dc1a18

tdonohue commented:
@kshepherd : I hope you don't mind, but I added a test for you in https://github.com/DSpace/DSpace/pull/8901/commits/804fce1c99419b29e745119460f3c351a8dc1a18.  Had a few moments this morning and figured out the best place to test this new adjustment.  The newly added test *fails* on `main` (as the Bundle permissions check fails)... but it should *succeed* in this PR.

tdonohue reviewed (approved):
👍 Thanks @kshepherd !  This looks correct to me & aligns with the discussion in the initial ticket.  As noted above, I added a test to prove it is working... that tests fails on `main`, but succeeds here.

tdonohue committed:
Fix to adjusting bundle and bitstream policies (item install, embargo lift)

Apply DEFAULT_ITEM_READ to bundles, not DEFAULT_BITSTREAM_READ so that
files can be listed if the item / default item is readable

tdonohue committed:
Fix to adjusting bundle and bitstream policies (item install, embargo lift)

Apply DEFAULT_ITEM_READ to bundles, not DEFAULT_BITSTREAM_READ so that
files can be listed if the item / default item is readable
Co-authored-by: kshepherd <kim@shepherd.nz>
Co-authored-by: tdonohue <tim.donohue@lyrasis.org>
